mobile/components/WithCurrentUser/index.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import * as React from 'react';
+import compose from 'recompose/compose';
+import {
+  getCurrentUser,
+  type GetUserType,
+} from '../../../shared/graphql/queries/user/getUser';
+
+type Props = {
+  render: Function,
+  data: {
+    user?: GetUserType,
+  },
+};
+
+class WithCurrentUser extends React.Component<Props> {
+  render() {
+    return this.props.render({ currentUser: this.props.data.user });
```

Comment:
Overall this feels pretty great from reading the code, how'd it feel writing it?

Nitpick: let's make this accept either `children` or `render`: `this.props.children ? this.props.children({}) : this.props.render({})`